### PR TITLE
Use core-layout

### PR DESCRIPTION
**Pull request by alex-ju**
Licence finder is the last standing frontend application (aside from `static`) that uses assets from `govuk_frontend_toolkit` and the only remaining one using what we call 'the `wrapper` layout' or ['wrapper slimmer template'](https://github.com/alphagov/static/blob/master/docs/slimmer_templates.md#wrapper-deprecated). 

Since we're not actively working on this app, but we don't want this dependency to stop us from dropping `govuk_frontend_toolkit` in `static`, we port the [required styles from static](https://github.com/alphagov/licence-finder/pull/914/files) to here (PR to remove them from static coming up) and map a few variables so we don't need to call them from `govuk_frontend_toolkit`. Note that `helpers/mixins` and `helpers/wrapper` are not ported as they are [made available to this app as part of the `core_layout`](https://github.com/alphagov/static/blob/master/app/assets/stylesheets/core-layout.scss).

This change also allows us to turn legacy colour pallette off and use new colours.

[Trello card](https://trello.com/c/Dpe7MWQp/2-frontend-applications-use-inconsistent-frontend-layouts)

<table>
<tr><th>Before</th><th>After</th></tr>
<tr><td valign="top">

Screenshot 2021-01-29 at 15 59 29

</td><td valign="top">

Screenshot 2021-01-29 at 16 00 18

</td></tr>
<tr><td valign="top">

www gov uk_licence-finder_sectors

</td><td valign="top">

www integration publishing service gov uk_licence-finder_sectors_cachebust=29Jan

</td></tr>
<tr><td valign="top">

www gov uk_licence-finder_location_sectors=3 activities=183

</td><td valign="top">

www integration publishing service gov uk_licence-finder_location_sectors=3 activities=183 cache=12

</td></tr>
<tr><td valign="top">

www gov uk_licence-finder_licences_activities=183 location=england sectors=3

</td><td valign="top">

www integration publishing service gov uk_licence-finder_licences_activities=183 location=england sectors=3 cachebust=123

</td></tr>
</table>

### CODE REVIEW

File: app/assets/stylesheets/_conditionals.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Media query helpers. These make producing IE layouts
+// super easy.
+
+// These are desktop and down media queries
+
+// There is also a local version of this in Smartanswers.
+
+$is-ie: false !default;
+
+@mixin media-down($size: false, $max-width: false, $min-width: false) {
```
Comment (andysellick) [inline app/assets/stylesheets/_conditionals.scss:10]: This mixin appears to be in the components gem already, although it doesn't include the `ie-lte` and `ie` mixins further down. Can this mixin be removed?

https://github.com/alphagov/govuk_publishing_components/blob/master/app/assets/stylesheets/govuk_publishing_components/components/mixins/_media-down.scss#L10

File: app/assets/stylesheets/_conditionals.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      @media (max-width: 800px) {
+        @content;
+      }
+    }
+  }
+}
+
+// mixins from toolkit, only used now by static and whitehall
+
+@mixin ie-lte($version) {
```
Comment (andysellick) [inline app/assets/stylesheets/_conditionals.scss:26]: Now looking a bit deeper... I can't see where this mixin is used, is it needed?

File: app/assets/stylesheets/_core.scss [hunk 3]
```diff
@@ -0,0 +1,153 @@
+// Core global styles
```
Comment (andysellick) [inline app/assets/stylesheets/_core.scss:1]: It's probably out of scope for this PR, but while much of this file is obviously needed I think we should consider switching out the base styles (`p` and headers) for components where possible. Same for some of the styles elsewhere. But probably for a followup PR.

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -14,6 +14,24 @@ $govuk-compatibility-govuktemplate: true;
 @import 'govuk_publishing_components/components/select';
 @import 'govuk_publishing_components/components/step-by-step-nav-header';
 
+// Map a few legacy variables to `govuk-frontend` equivalents
+// to avoid introducing a dependency on govuk_frontend_toolkit
+$text-colour: $govuk-text-colour;
+$secondary-text-colour: $govuk-secondary-text-colour;
+$link-visited-colour: $govuk-link-visited-colour;
```
Comment (andysellick) [inline app/assets/stylesheets/application.scss:21]: It looks like this is only used in one place (`publisher.scss`) so could this line be removed and use `govuk-link-visited-colour` directly there instead?

Comment (andysellick) [reply app/assets/stylesheets/application.scss:21]: Same with `link-hover-colour`, `link-colour` and `canvas-colour`.

File: app/assets/stylesheets/_conditionals.scss (hunk 1 above)
Comment (alex-ju) [reply app/assets/stylesheets/_conditionals.scss:10]: I haven't looked inside the SCSS files – it was more of a lift and shift operation without any improvements. This project is in discussions to be deprecated and I don't think it's worth investing time to improve it unless there's a mission dedicated to this. If I start looking inside the SCSS files I will most probably end up refactoring it all.

Comment (andysellick) [reply app/assets/stylesheets/_conditionals.scss:10]: That's fair enough then.

**Review by andysellick — APPROVED** [review]
On the basis that this PR does what it says it does, I'm going to approve it, but leave my comments about the SCSS structure in place for future reference. If we do retire this application, that's fine, but we end up keeping it and in any way need to further modify it, we should look at cleaning the frontend code a little.

**Comment by alex-ju** [comment]
Thanks Andy! I did a shallow clean-up on unused mixins. In the very unlikely case in which this project gets a revamp I hold myself up to rewrite this ported SCSS files.

File: app/assets/stylesheets/_conditionals.scss (hunk 2 above)
Comment (alex-ju) [reply app/assets/stylesheets/_conditionals.scss]: Removed

**Pull request merged**